wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/OmitWireEnumValueJsonAdapter.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import java.io.IOException
+
+private class OmitWireEnumValueJsonAdapter<T : WireEnum>(
```

Comment:
Rather than composing Moshi’s built-in enum adapter would you consider just having a WireEnumJsonAdapter? That way we get the behavior we want, regardless of what Moshi’s default does. In particular, the default one is configurable, and I don’t like thinking about what happens if it’s configured to do something weird, like lowercase enum constants.